internal/app/machined/pkg/system/services/kubelet.go

```diff
@@ -225,47 +232,99 @@ func (k *Kubelet) HealthSettings(runtime.Configurator) *health.Settings {
 	return &settings
 }
 
+func newKubeletConfiguration(clusterDNS []string) *kubeletconfig.KubeletConfiguration {
+	f := false
+	t := true
+
+	return &kubeletconfig.KubeletConfiguration{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "kubelet.config.k8s.io/v1beta1",
+			Kind:       "KubeletConfiguration",
+		},
+		StaticPodPath:      "/etc/kubernetes/manifests",
+		Address:            "0.0.0.0",
+		Port:               10250,
+		ReadOnlyPort:       10255, // TODO(andrewrynhard): Disable this.
+		RotateCertificates: true,
+		Authentication: kubeletconfig.KubeletAuthentication{
+			X509: kubeletconfig.KubeletX509Authentication{
+				ClientCAFile: constants.KubernetesCACert,
+			},
+			Webhook: kubeletconfig.KubeletWebhookAuthentication{
+				Enabled: &t,
+			},
+			Anonymous: kubeletconfig.KubeletAnonymousAuthentication{
+				Enabled: &f,
+			},
+		},
+		Authorization: kubeletconfig.KubeletAuthorization{
+			Mode: kubeletconfig.KubeletAuthorizationModeWebhook,
+		},
+		ClusterDomain:       "cluster.local",
+		ClusterDNS:          clusterDNS,
+		SerializeImagePulls: &f,
+		FailSwapOn:          &f,
+	}
+}
```

Comment:
This is basically a translation of old flags into their respective config field.